### ISSUE TITLE
[Pre Review] Making shot noise realizations reproduceable

### ISSUE THREAD

**Issue by Jashcraf** [UCSB-Exoplanet-Polarimetry-Lab/GRaTeR-JAX#215]
Saw the following in the Spline SPF tutorial regarding using actual shot noise realizations for the demo to motivate why $\sqrt{I}$ was used for the error map.

> It gives the optimiser a realistic signal-to-noise weighting without introducing stochastic variation that would make the notebook non-reproducible.

My understanding is that the random number generators used by modules like numpy and jax are only pseudo-random, partially meaning that if you use the same seed the same array will always be generated. One of Jax's "sharp bits" is that its API departs from numpy. In numpy, you do the following

```python
import numpy as np
seed = 24601
np.random.seed(seed)
sample_normal = np.random.normal()
```

Which will always produce the same random number. With older Jax versions, you need to

```python
import jax
import jax.numpy as np

seed = 04212026
key = jax.random.PRNGKey(seed)
sample_normal = np.random.normal(key=key)
```

I think newer Jax versions use `random.key`
```python
import jax
import jax.numpy as np

seed = 04212026
key = jax.random.key(seed)
sample_normal = np.random.normal(key=key)
```

So if you wanted to make a reproducible array drawn from a poisson distribution in the demo, you would

```python
import jax
import jax.numpy as np

seed = 04212026
key = jax.random.key(seed)
sample_normal = np.random.normal(key=key, mean=np.abs(truth_image))
```

and it would produce the same array every time
